0.preprocessing_data/scripts/3.perform_file_corruption_checks.py

```diff
@@ -0,0 +1,165 @@
+        strict=True
+    )
+    well_fov_dir = raw_images_path / well_fov
+    channel_images = {
+        channel_name: {"filename": [], "filepath": []} for channel_name in channel_names
+    }
+    for filename in well_fov_dir.glob("*"):
+        if filename.suffix in image_extensions:
+            for channel_name in channel_names:
+                if channel_name in filename.name:
+                    channel_images[channel_name]["filepath"].append(filename)
+
+    for channel_name in tqdm.tqdm(
+        channel_names, desc=f"Processing channels in {well_fov_dir.name}", leave=False
+    ):
+        channel_images[channel_name]["filepath"] = sorted(
+            channel_images[channel_name]["filepath"]
+        )
+        if not channel_images[channel_name]["filepath"]:
+            print(
+                f"No files found for channel {channel_name} in {well_dir}. Skipping..."
+            )
+            continue
+
+        images_to_stack = np.array(
+            [
+                tiff.imread(filepath)
+                for filepath in channel_images[channel_name]["filepath"]
+            ]
+        )
+        filepath = channel_images[channel_name]["filepath"][0]
+        well = str(filepath.parent).split("/")[-1]
+        output_path = zstack_output_path / f"{well}" / f"{well}_{channel_name}.tif"
+        output_path.parent.mkdir(exist_ok=True, parents=True)
+        print(
+            f"Saving max projection for {channel_name} channel in {well_fov} to {output_path}"
+        )
+        tiff.imwrite(output_path, images_to_stack)
+
+
+# ## Set input and output directories
+
+# In[3]:
+
+
+list_of_patients = [  # will be in a separate file in the future
+    "NF0014",
+    "NF0016",
+    "NF0018",
+    "NF0021",
+    "NF0030",
+    "NF0040",
+    "SARCO219",
+    "SARCO361",
+]
+
+
+# In[4]:
+
+
+patient_input_dict = {}
+for patient in list_of_patients:
+    patient_input_dict[patient] = {
+        "raw_images": pathlib.Path(f"../../data/{patient}/raw_images").resolve(),
+        "zstack_output": pathlib.Path(f"../../data/{patient}/zstack_images").resolve(),
+    }
+pprint.pprint(patient_input_dict)
+
+
+# ## Create list of the well-site folders
+
+# In[5]:
+
+
+patient_well_fovs_to_fix = []
+# loop through patients, well_fovs, and each channel
+# outer loop through patients
+for patient in tqdm.tqdm(patient_input_dict.keys(), desc="Processing patients"):
+    input_dirs = [
+        x for x in patient_input_dict[patient]["zstack_output"].iterdir() if x.is_dir()
+    ]
+    input_dirs.sort()
+
+    for well_dir in tqdm.tqdm(
+        input_dirs, desc=f"Processing wells for patient {patient}", leave=False
+    ):
+        # get the list of files in the well directory
+        files = list(well_dir.iterdir())
+        files.sort()
+        # make sure there are 5 files in the well directory
+        if len(files) != 5:
+            patient_well_fovs_to_fix.append(f"{patient} {well_dir.name}")
```

Comment:
OH okay so this is checking if the z-stack file is corrupted or not! So this must come after step 2. I would make this more clear I thought this was on the raw file folder prior to z-stack.